aten/src/ATen/native/quantized/cpu/qnnpack_maxpool.cpp

```diff
@@ -0,0 +1,145 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <aten/src/ATen/native/Pool.h>
+
+#include "init_qnnpack.h"
+#include "qnnpack_utils.h"
+
+namespace at {
+namespace native {
+namespace {
+
+class QNNPACKMaxPool final : public torch::OperatorKernel {
+ public:
+#ifdef USE_QNNPACK
+  Tensor operator()(
+      Tensor input,
+      std::vector<int64_t> kernel_size,
+      std::vector<int64_t> stride,
+      std::vector<int64_t> padding,
+      std::vector<int64_t> dilation) {
+    Tensor qy;
+
+    TORCH_CHECK(
+        input.ndimension() == 4,
+        "qnnpack_maxpool(): Expected input to be 4-dimensional: got ",
+        input.ndimension());
+    TORCH_CHECK(
+        kernel_size.size() == 2,
+        "qnnpack_maxpool(): Expected kernel_size to be 2-dimensional: got ",
+        kernel_size.size());
+    TORCH_CHECK(
+        stride.size() == 2,
+        "qnnpack_maxpool(): Expected stride to be 2-dimensional: got ",
+        stride.size());
+    TORCH_CHECK(
+        dilation.size() == 2,
+        "qnnpack_maxpool(): Expected dilation to be 2-dimensional: got ",
+        dilation.size());
+    TORCH_CHECK(
+        padding.size() == 2,
+        "qnnpack_maxpool(): Expected padding to be 2-dimensional: got ",
+        padding.size());
+
+    Tensor input_contig = input.contiguous();
+
+    initQNNPACK();
+    const auto scale = input_contig.q_scale();
+    const auto zero_point = input_contig.q_zero_point();
+    qnnp_operator_t qnnpackOperator_{nullptr};
+
+    int64_t padL = padding[0];
```

Comment:
Maybe rename to `padW` and `padH` to reflect the fact that left/right and top/bottom are symmetrical?

Also, W first when all other args are H first?